Moonshot/player/player.gd

```diff
@@ -86,6 +74,39 @@ func _physics_process(_delta) -> void:
 
 		cooldown = false
 
+
+func flip_facing() -> void:
+	facing *= -1
+	$AnimatedSprite.flip_h = !$AnimatedSprite.flip_h
+	
+	
+func set_facing(dir) -> void:
+	if sign(dir) == 0:
+		pass
+	elif sign(dir) != sign(facing):
+		flip_facing()
+
+
+func set_animation() -> void:
+	var new_state = state_machine.get_animation_name()
+	if new_state != null and new_state != animation_name:
+		animation_name = new_state
+		$AnimatedSprite.play(animation_name)
+
+
+func face_mouse() -> void:
+	var mouse_side := get_global_mouse_position().x - get_global_position().x
```

Comment:
You can replace `get_global_position().x` with  just `global_position.x` if you want.